app/controllers/api/v1/projects_controller.rb

```diff
@@ -37,6 +37,31 @@ def create
         end
       end
 
+      def key
+        if params[:id].nil? || params[:contribution_key].nil?
+          respond_to do |format|
```

Comment:
This line:
```
respond_to do |format|
```
is in every terminating condition of your code.

It would be more efficient to wrap ALL of you if statements (and else statements) inside a single one of these. That way, you only have that line once.